source/PlayState.hx

```diff
@@ -49,7 +62,37 @@ class PlayState extends FlxState
 	override public function update(elapsed:Float):Void
 	{
 		super.update(elapsed);
-		hud.update(elapsed);
+
 		FlxG.collide(map, player);
+
+		// If platform and player are not touching, allow offset to be overwritten
+		if (!FlxG.overlap(player, platform)) {
+			platform.sticky = false;
+		}
+
+		if (FlxG.collide(player, platform)) {
+			// Only set offset value if touching platform for "first" time
+			player.acceleration.y = 8000;
+			if (!platform.sticky) {
+				platform.sticky = true;
+				platform.offsetX = player.x - platform.x;
+			}
+			
+			// Multiply velocity by elapsed to get the player's movement each frame.
+			platform.offsetX += player.velocity.x * elapsed;
+			
+			// Allow player to drop through platform if down key pressed
+			if (FlxG.keys.anyPressed([FlxKey.DOWN])) {
+				player.y = platform.y;
```

Comment:
I don't feel like this is the best way to make the player drop through a platform, because it jumps them down to being beneath the platform in a single frame.

Instead, I think you could only check for a collision between player and platform if down is **not** held, which means the player will just naturally drop through the platform if the player holds down.